mods/page_save.js

```diff
@@ -124,16 +165,20 @@ PageSave.prototype.saveHtml = function(restbase, req) {
     var self = this;
     var rp = req.params;
     var title = rbUtil.normalizeTitle(rp.title);
-    var promise = P.resolve({});
     this._checkParams(req.body);
     // First transform the HTML to wikitext via the parsoid module
+    var path = [rp.domain, 'sys', 'parsoid', 'transform', 'html', 'to', 'wikitext', title];
+    var baseRevision = this._getBaseRevision(req);
+    if (baseRevision) {
+        path.push(baseRevision);
+    }
     return restbase.post({
-        uri: new URI([rp.domain, 'sys', 'parsoid', 'transform', 'html', 'to', 'wikitext', title]),
+        uri: new URI(path),
```

Comment:
This should also set `headers: { 'if-match': req.body.base_etag || headers['if-match'] }`, so that the html2wt end point can use that instead of the meta tag hack.